datafusion/core/src/execution/context.rs

```diff
@@ -106,6 +106,43 @@ use super::options::{
     AvroReadOptions, CsvReadOptions, NdJsonReadOptions, ParquetReadOptions, ReadOptions,
 };
 
+/// DataFilePaths adds a method to convert strings and vector of strings to vector of URLs.
+/// This is primarily used as restriciton on the input to methods like read_csv, read_avro
+/// and the rest.
+pub trait DataFilePaths {
+    /// Parse to a list of URLs
+    fn to_urls(self) -> Result<Vec<ListingTableUrl>>;
+}
+
+impl DataFilePaths for &str {
```

Comment:
technically speaking I think this might be a breaking change as the methods used to take `impl AsRef<str>` 

However, I think it is fine for this case as there is an easy workaround (call `as_ref()`)

Reply:
I'm trying to implement for AsRef<str> along with an iterator. I'll update once it's ready.

Reply:
Looks like rust won't allow it. We won't be able to implement a trait for another trait and also for another class.